### ISSUE TITLE
Extra Parentheses wrongly generated

### ISSUE THREAD

**Issue by Saheb** [scalameta/paradise#141]
```
class addFinal extends scala.annotation.StaticAnnotation {
  inline def apply(defn: Any): Any = meta {
    defn match {
      case something @ q"..$mods class $tname[..$tparams] ..$ctorMods (...$paramss) extends $template" =>
      val finalMod = collection.immutable.Seq(Final()) ++ mods
      q"..$finalMod class $tname[..$tparams] ..$ctorMods (...$paramss) extends $template"
    }
  }
}
```

```
 @addFinal
 case class Person(v: String) extends Human
```

`final case class Person(v: String) extends Human()`

The extra parentheses on Human shouldn't be present!

```
@ q"..$finalMod class $tname[..$tparams] ..$ctorMods (...$paramss) extends $template"
res4: Defn.Class = final case class Person(name: String) extends Human
```

**Comment by xeno-by** [comment]
Are we talking IntelliJ integration here?

**Comment by olafurpg** [comment]
The parentheses come from the converter

```
// original
@Main
object MainExample extends B {
  println("Hello scala.meta macros!")
}
// converted
object MainExample extends B() { println("Hello scala.meta macros!") }
```

**Comment by xeno-by** [comment]
I'm trying to understand why it matters whether the parentheses are there or not. Both alternatives (with and without parentheses) are semantically equivalent. Moreover, macro expansions never get prettyprinted except for IntelliJ support.

**Comment by Saheb** [comment]
From end user perspective, not getting the same thing what was initially present is a bit misleading. Apart from that as previously stated, they are same and compiles fine.

**Comment by DavidDudson** [comment]
I cannot seem to reproduce this at all. 

```
syntactic("class X extends Y")
syntactic("class X extends Y {}")
syntactic("class X extends Y { println(\"foo\") }")
```

I'm presuming this was fixed in #144
